vendor/processors/authorizenet.py

```diff
@@ -203,80 +202,104 @@ def create_billing_address(self):
+        order.invoiceNumber = str(self.invoice.pk)
+        order.description = self.get_transaction_id()
+
+        return order
+
     def get_payment_occurrences(self, subscription, subscription_type):
-        if subscription_type == TermType.SUBSCRIPTION:
-            return subscription.offer.term_details['payment_occurrences']
-        elif subscription_type == TermType.MONTHLY_SUBSCRIPTION:
-            return 1
-        elif subscription_type == TermType.QUARTERLY_SUBSCRIPTION:
-            return 1
-        elif subscription_type == TermType.SEMIANNUAL_SUBSCRIPTION:
-            return 1
-        elif subscription_type == TermType.ANNUAL_SUBSCRIPTION:
-            return 1
+        """
+        Gets the defined payment ocurrences for a Subscription. It defaults to
+        9999 which means it will charge that amount until the customer cancels the subscription. 
+        """
+        return subscription.offer.term_details.get('payment_occurrences', 9999)
 
     def get_period_length(self, subscription, subscription_type):
         if subscription_type == TermType.SUBSCRIPTION:
             return subscription.offer.term_details['period_length']
         elif subscription_type == TermType.MONTHLY_SUBSCRIPTION:
             return 1
         elif subscription_type == TermType.QUARTERLY_SUBSCRIPTION:
-            return 4
+            return 3
         elif subscription_type == TermType.SEMIANNUAL_SUBSCRIPTION:
             return 6
         elif subscription_type == TermType.ANNUAL_SUBSCRIPTION:
             return 12
 
-    def get_trail_occurrences(self, subscription):
+    def get_trial_occurrences(self, subscription):
         return subscription.offer.term_details.get('trial_occurrences', 0)
 
+    def get_payment_schedule_start_date(self, subscription):
+        """
+        Determines the start date offset so the paymente gateway starts charging the monthly subscriptions
+        """
+        units = subscription.offer.term_details.get('term_units', TermDetailUnits.MONTH)
+        if units == TermDetailUnits.MONTH:
+            return self.get_future_date_months(timezone.now(), self.get_trial_occurrences(subscription))
+        elif units == TermDetailUnits.DAY:
+            return self.get_future_date_days(timezone.now(), self.get_trial_occurrences(subscription))
+
     def create_payment_scheduale_interval_type(self, subscription, subscription_type):
         """
         Create an interval schedule with fixed months.
         period_length: The period length the service paided mor last 
             eg: period_length = 2. the user will be billed every 2 months.
         payment_occurrences: The number of occurrences the payment should be made.
             eg: payment_occurrences = 6. There will be six payments made at each period_length.
-        trial_occurrences: The number of ignored payments out of the payment_occurrences
+        trial_occurrences: The number of ignored payments out of the total payment occurrences
```

Comment:
yes that is actually what is happening. depending on the offer term_details on term_units and trail_occurrences you would have a different billing start date.
Eg.
1.  unit is days and trail_occurrences is 7. It would start charging the customer on the seventh day.
2. Unit is Month and Trail_occurrences is 1. it would start charging one month after.

Reply:
Do we need a separate trial interval option, that defaults to the same interval unit as the offer?

Reply:
Do we need a separate trial interval option, that defaults to the same interval unit as the offer?